### PR TITLE
Add transaction hook for global clock(#24519)

### PR DESCRIPTION
**Pull request by FlyingZC**
For #24519 .

Changes proposed in this pull request:
  - Add transaction hook for global clock

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: kernel/global-clock/core/src/main/java/org/apache/shardingsphere/globalclock/core/executor/GlobalClockTransactionHook.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.Properties;
+
+/**
+ * Global clock transaction hook.
+ */
+public class GlobalClockTransactionHook extends TransactionHookAdapter {
```
Comment (zhaojinchao95) [inline kernel/global-clock/core/src/main/java/org/apache/shardingsphere/globalclock/core/executor/GlobalClockTransactionHook.java:37]: Add `final` please.

File: kernel/global-clock/core/src/main/java/org/apache/shardingsphere/globalclock/core/executor/OpenGaussGlobalClockTransactionExecutor.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Collection;
+
+/**
+ * OpenGauss global clock transaction executor.
+ */
+public class OpenGaussGlobalClockTransactionExecutor implements GlobalClockTransactionExecutor {
```
Comment (zhaojinchao95) [inline kernel/global-clock/core/src/main/java/org/apache/shardingsphere/globalclock/core/executor/OpenGaussGlobalClockTransactionExecutor.java:28]: Add `final` please

File: kernel/transaction/api/src/main/java/org/apache/shardingsphere/transaction/spi/TransactionHookAdapter.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import org.apache.shardingsphere.sql.parser.sql.common.enums.TransactionIsolationLevel;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Collection;
+
+/**
+ * ShardingSphere transaction hook adapter.
+ */
+public class TransactionHookAdapter implements TransactionHook {
```
Comment (zhaojinchao95) [inline kernel/transaction/api/src/main/java/org/apache/shardingsphere/transaction/spi/TransactionHookAdapter.java:31]: Add `final` please

File: kernel/global-clock/core/src/main/java/org/apache/shardingsphere/globalclock/core/rule/GlobalClockRule.java [hunk 4]
```diff
@@ -40,12 +46,26 @@ public final class GlobalClockRule implements GlobalRule {
...
         configuration = ruleConfig;
         enabled = ruleConfig.isEnabled();
         globalClockProvider = enabled ? TypedSPILoader.getService(GlobalClockProvider.class, String.join(".", ruleConfig.getType(), ruleConfig.getProvider()),
                 null == ruleConfig.getProps() ? new Properties() : ruleConfig.getProps()) : null;
-        Optional.ofNullable(globalClockProvider).ifPresent(optional -> optional.init(ruleConfig.getProps()));
+        TypedSPILoader.getService(TransactionHook.class, "GlobalClock", getProps(ruleConfig, databases));
+    }
+    
+    private Properties getProps(final GlobalClockRuleConfiguration ruleConfig, final Map<String, ShardingSphereDatabase> databases) {
+        Properties result = new Properties();
```
Comment (strongduanmu) [inline kernel/global-clock/core/src/main/java/org/apache/shardingsphere/globalclock/core/rule/GlobalClockRule.java:58]: You can use PropertiesBuilder to reduce code.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: kernel/global-clock/core/src/main/java/org/apache/shardingsphere/globalclock/core/rule/GlobalClockRule.java (hunk 4 above)
Comment (FlyingZC) [reply kernel/global-clock/core/src/main/java/org/apache/shardingsphere/globalclock/core/rule/GlobalClockRule.java:58]: PropertiesBuilder just in test scope now.

**Pull request merged**